packages/x-date-pickers/src/internals/hooks/useViews.tsx

```diff
@@ -9,6 +9,11 @@ export type PickerOnChangeFn<TDate> = (
   selectionState?: PickerSelectionState,
 ) => void;
 
+export type NonNullablePickerChangeHandler<TDate> = (
```

Comment:
I will rename `PickerOnChangeFn` `NullablePickerChangeHandler` in a follow up PR